keps/sig-testing/2539-continuously-deploy-k8s-prow/README.md

```diff
@@ -0,0 +1,149 @@
+
+## Motivation
+
+Currently, deploying k8s prow consists of following steps:
+
+1. Updates made in prow are released as container images.
+1. Automated process creates PRs updating the prow images tags.
+1. Oncall inspects prow logs to make sure prow is safe to upgrade.
+1. Oncall inspects PRs created in step #2, approve them, and post message on Slack.
+1. Oncall waits until #4 and a postsubmit job applying the changes on prow cluster.
+1. Oncall waits until #5 is done, do several manual inspections to make sure prow works.
+
+This is a very time consuming process (Roughly 3 hours per week for oncall), especially the context switching between the waits makes oncall hard to focus on their day job. Thus it’s desired to streamline the process so that we can save time.
+
+Historically, the biggest reasons why the manual processes were introduced are:
+
+- Errors in prow were not easily discoverable, most of time were reported by prow users, which was bad.
+
+This problem has been largely solved by the introduction of prow monitoring + alerting by grafana, prometheus, and prometheus alertmanager stack. And based on our experience in the past quarter, we haven’t had a single case where prow errors were discovered by humans earlier than by prow alerts. This fact gives us reasonable confidence to proceed with continuous delivery, and the following assumptions should hold:
+
+- Prow is stable as long as there is no alert. (Indicate that no need to inspect prow logs before bumping)
+- Errors caused by prow upgrades are discovered by prow alerts in a timely manner.
+
+
+### Goals
+
+The proposed change switches from daily manual deployment to hourly automated deployment.
+
+#### Prow Users
+
+Shouldn’t see any change, prow breakage should be discovered by prow monitoring system and rollback will be performed. The chance of prow being break is almost identical to what we have today(Assume there are not more than a single breaking change every day).
+
+#### Prow Oncall
+
+- What’s Not Changed
+  - React to prow alerts and take actions.
+- What’s Changed
+  - No more manual inspecting prow healthiness.
+  - No more manual lgtm/approve/retest autobump PRs.
+  - No more manual Slack posting.
+
+
+### Non-Goals
+
+Change how prow is released.
+
+
+## Proposal
+
+Prow autobump PRs are automatically merged every hour, only on working hours of working days.
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Breaking Changes in Prow
+Breaking changes in prow will require manual intervention. Currently prow isn’t able to handle these intelligently, as it was not designed with the mindset of API versions and thus kubernetes conversion webhook can not help coping with breaking changes among major APIs.
+One possible way of dealing with breaking changes, is:
+- Prow oncall inspects prow logs and breaking changes announcements once per week, and take actions based on deprecation warnings from prow logs and breaking changes from ANNOUNCEMENTS.md.
```

Comment:
is this sufficient if we're merging hourly?

Reply:
Good point, as mentioned in https://github.com/kubernetes/enhancements/pull/2540/files#r586723114, we'll start with 3 hours, and we plan to rely on alerts instead of oncall scanning log for prow errors discovery